pkg/cloud/services/secretsmanager/secret.go

```diff
@@ -76,12 +76,12 @@ func (s *Service) Create(m *scope.MachineScope, data []byte) (string, int32, err
 }
 
 // Delete the secret belonging to a machine from AWS Secrets Manager
-func (s *Service) Delete(m *scope.MachineScope) error {
+func (s *Service) Delete(prefix string, chunkCount int32) error {
```

Comment:
Is this just a style preference? I'm not sure what is the benefit w.r.t. the linked issue

Reply:
This isn't a style change. It's tied to not actually saving the secret prefix if it failed to create, so it can get retried, but I still want to make one attempt at rolling back the secrets.

That said, I might try to cut this differently.

/hold